src/main/factories/validations/change-password-validation-factory.ts

```diff
@@ -0,0 +1,26 @@
+import { type Validation } from '@/application/protocols'
+import {
+  ValidationComposite,
+  RequiredFieldValidation,
+  PasswordValidation,
+  CompareFieldsValidation
+} from '@/application/validation'
+import { PasswordValidatorAdapter } from '@/infra/validators'
+import { PasswordMismatchError, PasswordRequirementsError } from '@/application/errors'
+
+export const makeChangePasswordValidation = (): ValidationComposite => {
+  const validations: Validation[] = []
+  const requiredFields = ['userId', 'password', 'passwordConfirmation']
+
+  for (const field of requiredFields) {
+    validations.push(new RequiredFieldValidation(field))
+  }
+
+  const passwordMismatch = (): Error => new PasswordMismatchError()
+  validations.push(new CompareFieldsValidation('password', 'passwordConfirmation', passwordMismatch))
```

Comment:
Just for clarification. Why this errors are here? I understood that this arguments errors are custom erros, so PasswordMismatchError is a specific error for this case? If you don't use this argument, a generic error will be returned by this validator, right?

Reply:
yes, exactly that. I created these generic errors to set the error messages requested in the user story.